### PR TITLE
Ensure runTx always returns details about the transaction in both VM and Web3 mode

### PR DESCRIPTION
**Pull request by axic**
Basically this merges `tryTillResponse()` into `runTx()` as I think it belongs there.

More testing and fine-tuning might be required.

### CODE REVIEW

File: src/universal-dapp.js [hunk 1]
```diff
@@ -481,31 +483,8 @@ UniversalDApp.prototype.getCallButton = function (args) {
...
-                    function testResult (err, address) {
-                        if (!err && !address) {
-                            setTimeout(function () { tryTillResponse(txhash, done); }, 500);
-                        } else {
-                            done(err, address);
-                        }
-                    }
-
-                }
-                tryTillResponse(result, function (err, result) {
```
Comment (redsquirrel) [inline src/universal-dapp.js]: Why did you remove this call to `tryTillResponse`? It seems like that would change the behavior (in a bad way).

Comment (axic) [inline src/universal-dapp.js]: `tryTillResponse` is merged into `runTx`

Comment (redsquirrel) [inline src/universal-dapp.js]: Aha, nice. I haven't tested it, but that seems like the right approach.

**Comment by axic** [comment]
Rebased (makes #43 much cleaner also)

**Comment by chriseth** [comment]
Change looks good, is it tested (also taking the recent merges into account)?

**Comment by axic** [comment]
Merge #79 first.

This was working yesterday, but wasn't tested with the #43 change.

**Comment by axic** [comment]
@redsquirrel if you have a chance, please try this.

It worked for me for both regular transactions and library deployments. It also decodes the response in web3 mode when calling constants (those which are calls and not transactions).

**Comment by redsquirrel** [comment]
@axic I tested this via web3 with a standalone contract, a standalone library, and a contract with a link to a library. It all deployed fine to morden. 👍
